### PR TITLE
citra-qt: Add an option to save the icon of a game

### PR DESCRIPTION
**Pull request by FearlessTobi**
I think the title is pretty self-explanatory. The option for this can be used by right-clicking the games entry in the compatibility list. The icon is then saved in the PNG image format. 
This feature is especially helpful for compatibility wiki submissions, as our current way of getting the games icon is not very user friendly.

Screenshot: 
screenshot

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3581)

### CODE REVIEW

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -774,6 +775,18 @@ void GMainWindow::OnGameListOpenFolder(u64 program_id, GameListOpenTarget target
     QDesktopServices::openUrl(QUrl::fromLocalFile(qpath));
 }
 
+void GMainWindow::OnGameListSaveIcon(QPixmap& game_icon) {
+    QString filepath = QFileDialog::getSaveFileName(this, tr("Save File"), "Icon.png",
```
Comment (CodingKoopa) [inline src/citra_qt/main.cpp]:
```cpp
tr("Icon.png")
```

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -774,6 +775,18 @@ void GMainWindow::OnGameListOpenFolder(u64 program_id, GameListOpenTarget target
...
+                                                    "PNG (*.png);;" + tr("All Files (*.*)"));
```
Comment (CodingKoopa) [inline src/citra_qt/main.cpp]:
```cpp
tr("PNG (*.png);;")
```

Comment (FearlessTobi) [reply src/citra_qt/main.cpp]: I'm okay with your suggestion above, but why does this need a "tr" statement? I don't know of any language where the file type PNG has another name.

File: src/citra_qt/game_list.cpp [hunk 3]
```diff
@@ -314,12 +314,14 @@ void GameList::PopupContextMenu(const QPoint& menu_location) {
     int row = item_model->itemFromIndex(item)->row();
     QStandardItem* child_file = item_model->invisibleRootItem()->child(row, COLUMN_NAME);
     u64 program_id = child_file->data(GameListItemPath::ProgramIdRole).toULongLong();
+    QPixmap game_icon = child_file->data(Qt::DecorationRole).value<QPixmap>();
```
Comment (BreadFish64) [inline src/citra_qt/game_list.cpp]: Should this be a reference?

**Comment by FearlessTobi** [comment]
I'm going to close this PR because jroweboy is strongly opposed to this change, at least in it's current design. Also there are plans to add all game icons to the list at once using https://github.com/Repo3DS/shop-cache.

**Pull request closed without merging**